spec/lib/pull_request_spec.rb

```diff
@@ -340,6 +397,28 @@ def create_mock_dependency_manager
       )
       pull_request.tell_dependency_manager_what_dependabot_is_changing
     end
+
+    it "only looks at the dependencies listed in the commit message" do
+      dependency_manager = double("DependencyManager")
+      api_response = "foo"
+      pull_request = PullRequest.new(api_response, dependency_manager)
+      allow(pull_request).to receive(:commit_message).and_return(multiple_dependencies_commit)
+      allow(pull_request).to receive(:gemfile_lock_changes).and_return(
+        <<~GEMFILE_LOCK_DIFF,
+          govuk_personalisation (0.13.0)
+                  plek (>= 1.9.0)
+                  rails (>= 6, < 8)
+          -    govuk_publishing_components (35.7.0)
+          +    govuk_publishing_components (35.8.0)
```

Comment:
Could you tweak this example diff to include `rack`, `rails` and `govuk_sidekiq` (in addition to `govuk_publishing_components`)?

Then have assertions something like:

```ruby
expect(dependency_manager).to receive(:add_dependency).with("rack")
expect(dependency_manager).to receive(:add_dependency).with("rails")
expect(dependency_manager).to receive(:add_dependency).with("govuk_sidekiq")
expect(dependency_manager).not_to receive(:add_dependency).with("govuk_publishing_components")
```